core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -161,7 +161,8 @@ private void configureProxy(Channel ch, ProxyConfig proxyConfig, SessionProtocol
 
         if (proxyConfig instanceof ConnectProxyConfig && ((ConnectProxyConfig) proxyConfig).useTls()) {
             final SslContext sslCtx = bootstraps.getOrCreateSslContext(proxyAddress, desiredProtocol);
-            ch.pipeline().addFirst(sslCtx.newHandler(ch.alloc()));
+            ch.pipeline().addFirst(sslCtx.newHandler(ch.alloc(), proxyAddress.getHostString(),
+                                                     proxyAddress.getPort()));
```

Comment:
They are necessary for hostname validation: https://github.com/netty/netty/wiki/Netty-4.2-Migration-Guide#:~:text=Ensure%20endpoint%20validation,in%20Netty%204.1.112.